src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> currentNode = getNodeByKey(key);
+        if (currentNode != null) {
+            currentNode.value = value;
+        } else {
+            if ((table == null ? 0 : table.length) * LOAD_FACTOR < size + 1) {
+                resize();
+            }
+            currentNode = new Node<>(key, value, null);
+            putNode(currentNode);
+            size++;
```

Comment:
Algorithm of put() method:
1. resize if need 
2. getIndex
3. create new node
4. check table by index - if null add new node here, change size and return 
5. get currentNode 
6. while currentNode =! null do 7, 8 and 9
7. if currentNode key equals key  - just rewrite value and return 
8. if currentNode.next == null - add new node as next, change size, return
9. currentNode = currentNode.next;